package.json

```diff
@@ -39,26 +39,32 @@
     "test": "tap test --coverage-report=lcov --no-browser"
   },
   "devDependencies": {
-    "metalsmith": "^2.3.0",
-    "rimraf": "^2.6.3",
-    "tap": "^12.5.3"
+    "metalsmith": "^2.5.1",
+    "tap": "^16.3.2"
   },
   "dependencies": {
-    "imagemin": "^6.1.0",
-    "imagemin-advpng": "^5.0.0",
-    "imagemin-giflossy": "^5.1.10",
-    "imagemin-gifsicle": "^6.0.1",
-    "imagemin-gm": "^2.0.1",
-    "imagemin-jpeg-recompress": "^6.0.0",
-    "imagemin-jpegoptim": "^6.0.0",
-    "imagemin-jpegtran": "^6.0.0",
-    "imagemin-mozjpeg": "^8.0.0",
-    "imagemin-optipng": "^6.0.0",
-    "imagemin-pngcrush": "^6.0.0",
-    "imagemin-pngout": "^4.0.0",
-    "imagemin-pngquant": "^7.0.0",
-    "imagemin-svgo": "^7.0.0",
-    "imagemin-webp": "^5.0.0",
-    "imagemin-zopfli": "^6.0.0"
+    "imagemin": "^7.0.1",
```

Comment:
the latest versions of these packages have all been switched to ESM modules, which won't work with this CJS module, hence why I kept them at the specific versions prior to ESM conversion...

what are your thoughts on ESM vs. CJS for Metalsmith?

Reply:
Not all. I tested and reviewed their GH pages one-by-one and upgraded to the latest CJS-compatible versions. I may have missed one. The number of security warnings metalsmith-imagemin gave was too high to ignore (76), let alone include in a build in a professional settting. It still has about 30 I think, but there is little you or me can do but use package.json `overrides` in each project.

Regarding ESM/CJS all core plugins are (on the way to be) distributed as dual ESM/CJS packages with TS support. I wrote [a news about it](https://metalsmith.io/news/2022-11-05/es-modules-typescript-and-new-docs/#dual-esmcjs-support-for-core-plugins) recently. Using dynamic import may be a better future alternative than using a bundler (because it is supported in both CJS/ESM environments). But CJS will be supported as long as metalsmith supports versions of Node where `{ "type": "module" }` is not the default (= all existing versions).

Reply:
FYI: getting npm install failures here...

![image](https://user-images.githubusercontent.com/183195/208749732-53eccc78-a314-4969-9f4c-da10a0f03801.png)

seems like `imagemin-advpng` does a postinstall step where it downloads a binary using `got`, one needs to delete `node_modules` folder and `package-lock.json` completely run `npm i` again after every change of those package versions just to make sure everything works...

doing so, shows the breakage with `got@v12.5.3` override.

unfortunately the latest non-esm version of got is [v11.8.6](https://github.com/sindresorhus/got/releases/tag/v11.8.6) which cannot be used in this override either, as it causes `jpeg-recompress` to fail pre-build step ...

![image](https://user-images.githubusercontent.com/183195/208751093-a1a592ce-381c-418e-acfa-456d219eb837.png)

... 

I think the only next step (to avoid all the vuln warnings) I could convert this whole package to a dual esm / cjs compiled package and make it work on both ...

Reply:
seems like imagemin is dead and we shouldn't expect anything to work in these dependencies ... so the ESM conversion is needed (but likely wont stop future vulns)

I'll continue the esm conversion, but then also look into replacing the internals with https://github.com/GoogleChromeLabs/squoosh

Reply:
I noticed and was writing a reply but forgot to hit comment:
The problem is the imagemin project itself is [actually unmaintained](https://github.com/imagemin/imagemin/issues/385), and [the lib it was based on](https://github.com/rflynn/imgmin) is unmaintained too. But the underlying binaries are still actively developed (jpegoptim, advpng etc).
So indeed Squoosh would be a better alternative, or forking the dependency chain of imagemin (for each plugin: imagemin-<plugin>, <plugin>-bin, bin-wrapper). A.o. this would allow configuring using a dynamically linked binary (instead of the ones shipped with imagemin plugins) which is suitable for custom devservers (but not for the likes of Netlify or Github Pages)

Still there's ~5 commits in this PR you could cherry-pick and apply for a final CJS-compatible release.